File: fleet/fleet_manager/report/vehicle_income_summary/vehicle_income_summary.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import msgprint, _

def execute(filters=None):
	if not filters: filters = {}

	advances_list = get_advances(filters)
	columns = get_columns()

	if not advances_list:
		msgprint(_("No record found"))
		return columns, advances_list

	data = []
	for advance in advances_list:
		row = [advance.name, advance.vehicle,advance.owner, advance.date, advance.amount]
		data.append(row)

	return columns, data


def get_columns():
	return [
		{
			"label": _("Title"),
			"fieldname": "title",
			"fieldtype": "Link",
			"options": "Vehicle Income",
			"width": 120
		},
		{
			"label": _("Vehicle"),
			"fieldname": "vehicle",
			"fieldtype": "Link",
			"options": "Vehicle",
			"width": 120
		},
		{
			"label": _("Owner"),
			"fieldname": "owner",
			"fieldtype": "Link",
			"options": "Customer",
			"width": 120
		},
		{
			"label": _("Received On"),
			"fieldname": "service_date",
			"fieldtype": "Date",
			"width": 120
		},
		{
			"label": _("Amount"),
			"fieldname": "amount",
			"fieldtype": "Currency",
			"width": 120
		}
	]

def get_conditions(filters):
	conditions = ""

	if filters.get("vehicle"):
		conditions += " and vehicle = %(vehicle)s"
	if filters.get("owner"):
		conditions += " and owner = %(owner)s"
	if filters.get("from_date"):
		conditions += " and date>=%(from_date)s"
	if filters.get("to_date"):
		conditions += " and date<=%(to_date)s"

	return conditions

def get_advances(filters):
	conditions = get_conditions(filters)
	return frappe.db.sql("""select name, vehicle, owner, date, amount , was_due_on
		from `tabVehicle Income`
		where docstatus<2 %s order by date ASC""" %
						 conditions, filters, as_dict=1)

File: fleet/fleet_manager/report/vehicle_income_summary/vehicle_income_summary.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Vehicle Income Summary"] = {
//    "onload":function(frm){
//        frappe.msgprint(__("jjjj"))
//    },
	"filters": [
		{
			"fieldname":"vehicle",
			"label": __("Vehicle"),
			"fieldtype": "Link",
			"options": "Vehicle",
			"width": "80"
		},
		{
			"fieldname":"from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"default": frappe.defaults.get_user_default("year_start_date"),
			"width": "80"
		},
		{
			"fieldname":"to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"default":frappe.datetime.add_days(frappe.datetime.get_today(), 2)
		},
		{
			"fieldname":"owner",
			"label": __("Owner"),
			"fieldtype": "Link",
			"options": "Customer"
		},
		{
			"fieldname":"status",
			"label": __("Status"),
			"fieldtype": "Select",
			"options": "\nPaid\nUnpaid",
			"default":"Paid"
		}
	],
	"formatter":function (row, cell, value, columnDef, dataContext, default_formatter) {
        value = default_formatter(row, cell, value, columnDef, dataContext);
        if (columnDef.id == "Amount")
            {
                if(dataContext.amount>"7000")
                    {
                        value = "<span style='color:green!important;font-weight:bold;'>" + value + "</span>";
                    }
                else if(dataContext.amount<"7000"){
                    value = "<span style='color:red!important;font-weight:bold;'>" + value + "</span>";
                }

            }
        return value;
    }

};




